src/genalg.cpp

```diff
@@ -130,25 +130,23 @@ void GenAlg::UpdateGenomeScores(const std::vector<double>& fitness_scores)
 
 void GenAlg::UpdateBestGenomes()
 {
+    m_best_genomes.clear();
     m_best_ever_fitness = std::max(m_best_ever_fitness, m_genomes[0].Fitness());
 
     for(int i = 0; i < NUM_BEST_GENOMES; ++i)
     {
-        m_best_genomes.push_back(&m_genomes[i]);
+        m_best_genomes.push_back(m_genomes[i]);
     }
 }
 
 void GenAlg::SpeciateGenomes()
```

Comment:
Remove all species after the aggregate amount of num spawned starts exceeding the population size.